client/src/components/formHelpers/form-validators.tsx

```diff
@@ -19,6 +19,27 @@ function isPathRoot(urlString: string): boolean {
 
 type Validator = (value: string) => React.ReactElement | null;
 
+// example link: https://learn.microsoft.com/en-us/training/achievements/learn.wwl.get-started-c-sharp-part-1.trophy?username=moT01&sharingId=E2EF453C1F9208B8
+export const isMicrosoftLearnLink = (value: string): boolean => {
+  let url;
+  try {
+    url = new URL(value);
+  } catch {
+    return false;
+  }
+
+  const correctDomain = url.hostname === 'learn.microsoft.com';
+  const correctPath = !!url.pathname.match(
+    /^\/[^/]+\/training\/achievements\//
+  );
+  const hasSharingId = !!url.searchParams.get('sharingId');
+  const hasUsername = !!url.searchParams.get('username');
```

Comment:
I'm fine with this how it is, but these two could potentially have a little more of a check - maybe `username.length > 0 && sharingId.length === 16` - Course, we don't know if all `sharingId`'s will be length 16, so maybe just > 0. We can leave it for now.

Reply:
That's what we have, but I can make it clearer by checking the length explicitly.

Reply:
No, not needed if it's already doing that. I guess I wasn't sure what something like `https://learn.microsoft.com/en-us/training/achievements/learn.wwl.get-started-c-sharp-part-1.trophy?username=&sharingId=` would give.